src/utils/bolt11.ts

```diff
@@ -18,8 +18,16 @@ export const decodeBolt11 = (invoice: string) => {
    const amountSection = findSection(decoded.sections, 'amount');
    const amountSat = amountSection?.value ? Number(amountSection.value) / 1000 : undefined;
 
-   const expiryUnixSeconds = decoded.expiry;
-   const expiryUnixMs = decoded.expiry * 1000;
+   const expirySection = findSection(decoded.sections, 'expiry');
+   const timestampSection = findSection(decoded.sections, 'timestamp');
 
-   return { amountSat, expiryUnixMs };
+   /* infinity if no expiry is defined */
+   let expiryUnixSec = Infinity;
+   if (expirySection && timestampSection) {
+      expiryUnixSec = timestampSection.value + expirySection.value;
+   } else if (decoded.expiry && timestampSection) {
```

Comment:
what is the diff between decoded.expiry and findSection(decoded.sections, 'expiry')

Reply:
They should be the same. The library says decoded.expiry is always a number, but some invoices had no expiry in either place. Due to the nature of open source being inconsistent in following specs I figure check for both to try our best to get it.

Reply:
I just checked the code and I think the previous code was correct. at least according to the source code

https://github.com/nbd-wtf/light-bolt11-decoder/blob/master/bolt11.js#L363

the source code is already summing those two sections

Reply:
Weird... that's not the behavior I'm getting

![image](https://github.com/user-attachments/assets/0773ffb5-fd69-4943-a12f-f9e600d6694c)
![image](https://github.com/user-attachments/assets/ad3f8615-3819-4fa0-be7d-e3a50c30eb82)

Reply:
weird indeed. can you debug and see what code gets executed in the lib?

Reply:
It looks like it calls [this block](https://github.com/nbd-wtf/light-bolt11-decoder/blob/master/bolt11.js#L379-L384) so it gets the expiry value directly. This seems like a bug in the libary. `expiry` is in `TAGCODES` so it gets overridden

Reply:
indeed. should we report it to them?

I would then just get rid of any usage of `decoded.expiry` in our code and just leave the approach with sections. we can move back to expiry if they ever fix it

Reply:
yea good point. I got rid of decoded.expiry, please check again